### DISCUSSION TITLE
building express app Docker image

### DISCUSSION THREAD

**Discussion by PriyanshuLakra** [vercel/turborepo#8534] [Help] [answered]
### Summary

I am working on a project in TurboRepo which contains two sub-projects: a Next.js application and a pure backend project. I have successfully created a Docker image for the Next.js application. However, I am encountering issues when trying to run the Docker image for my backend project.

Backend Project Specifications
Framework: Express
Language: TypeScript
Scripts in package.json:

"scripts": {
  "build": "npx esbuild ./src/index.ts --bundle --platform=node --outfile=dist/index.js",
  "start": "node dist/index.js",
  "dev": "npm run build && npm run start"
}

tsconfig.json Configuration
I have changed the tsconfig.json file to set:

"outDir": "./dist",
"rootDir": "./src"

Project Structure
The file structure of my project is as follows:
MyApp(turborepo)
|
|__apps
└── bank-webhook/
    ├── package.json
    ├── tsconfig.json
    └── src/
        └── index.ts

When running the backend project locally using npm run dev, it works as expected.

**Dockerfile**

I have created the following Dockerfile for the backend project:

# Use an official Node runtime as a parent image
FROM node:20.12.0-alpine3.19

# Set the working directory for the app
WORKDIR /apps/bank-webhook

# Copy the necessary files
COPY package.json tsconfig.json ./
COPY . .

# Install dependencies
RUN npm install

# Build the application
RUN npm run build

# Expose the port the app runs on
EXPOSE 3003

# Start the application
CMD ["npm", "run", "dev"]

**Issue Encountered**

I successfully build the Docker image using the command:  docker build -t mybackend .

However, when I try to run the image using the command: docker run -p 49160:3000 mybackend

I encounter the following error:

<img width="1079" alt="Screenshot 2024-06-19 at 2 37 43 AM" src="https://github.com/vercel/turbo/assets/139461877/4723fa04-2625-458c-b72a-e03e80b02969">

**Request for Assistance**

I would appreciate any guidance on resolving this issue. Specifically, I am looking for advice on:

1.) Verifying the Dockerfile configuration for building and running a TypeScript-based Express backend.
2.) Ensuring the correct ports are exposed and mapped.
3.) Any other potential issues or best practices for running this setup in Docker.

Thank you in advance for your help!

**Comment by anthonyshew** [comment, marked as answer]
The Docker guide looks like it would help here: https://turbo.build/repo/docs/guides/tools/docker

**Reply by PriyanshuLakra** [reply]
thankyou  Anthonyshew  for your help , it worked
